aiohttp/client_reqrep.py

```diff
@@ -428,11 +431,20 @@ def update_transfer_encoding(self) -> None:
             if hdrs.CONTENT_LENGTH not in self.headers:
                 self.headers[hdrs.CONTENT_LENGTH] = str(len(self.body))
 
-    def update_auth(self, auth: Optional[BasicAuth]) -> None:
+    def update_auth(self, auth: Optional[BasicAuth], trust_env: bool = False) -> None:
         """Set basic auth."""
         if auth is None:
             auth = self.auth
         if auth is None:
+            if trust_env:
+                # If no auth is specified, we read from netrc.
+
+                netrc_obj = netrc_from_env()
+                if netrc_obj:
+                    auth = basicauth_from_netrc(netrc_obj, self.url.host)
+                    if auth:
+                        self.headers[hdrs.AUTHORIZATION] = auth.encode()
```

Comment:
I'd be inclined not to repeat this code here. Instead doing the above code under `if auth is None and trust_env:` and having another `if auth is None: return` block afterwards...